azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -155,17 +155,12 @@ public Project fetchProjectByName(final String name) throws ProjectManagerExcept
     Project project = null;
     final ProjectResultHandler handler = new ProjectResultHandler();
 
-    // select active project from db first, if not exist, select inactive one.
     // At most one active project with the same name exists in db.
     try {
-      List<Project> projects = this.dbOperator
+      final List<Project> projects = this.dbOperator
           .query(ProjectResultHandler.SELECT_ACTIVE_PROJECT_BY_NAME, handler, name);
       if (projects.isEmpty()) {
-        projects = this.dbOperator
-            .query(ProjectResultHandler.SELECT_PROJECT_BY_NAME, handler, name);
-        if (projects.isEmpty()) {
-          throw new ProjectManagerException("No project with name " + name + " exists in db.");
-        }
+        throw new ProjectManagerException("No active project with name " + name + " exists in db.");
```

Comment:
should we just return null instead of throwing exception?

Reply:
changed, thanks.